astrology.lic

```diff
@@ -54,6 +54,11 @@ class Astrology
                      .values
                      .first
     buffs.reject! { |spell| spell.eql?('Read the Ripples') }
+
+    buffs.keys.each do |key|
```

Comment:
What do you think about using the hash directly instead of `.keys`? Something like this:
```ruby
    buffs.each do |key, value|
      check_discern(value, settings) if value['use_auto_mana']
    end
```

Reply:
What do you think about using method chaining for the `if`? Something like this:
```ruby
    buffs.select { |_key, value| value['use_auto_mana'] }
         .each { |_key, value| check_discern(value, settings) }
```

Reply:
can you explain this syntax with the if statement. It looks to me like that would be applied to everything following the statement than what is before the statement.

Reply:
The original code iterates over every item, but only runs `check_discern` if a condition is met.

The `.select` syntax only keeps items for which the condition is met, then runs `check_discern` over everything that was kept.

See https://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-select for the technical details.